src/repository-utils.ts

```diff
@@ -0,0 +1,143 @@
+    cwd: repositoryDirectoryPath,
+  });
+  const match = stdout.match(/^refs\/heads\/(.+)$/u);
+  const currentBranchName = match?.[1] ?? null;
+  if (!currentBranchName) {
+    throw new Error(
+      `The repository '${repositoryDirectoryPath}' does not seem to be on a branch. Perhaps HEAD is detached? Either way, you will need to return this repo to the default branch manually.`,
+    );
+  }
+  return currentBranchName;
+}
+
+/**
+ * Retrieves the default branch of the given repository, that is, the branch
+ * that represents the main line of development. Unfortunately there's no good
+ * way to obtain this, so this function just tries "main" followed by "master".
+ *
+ * @param repositoryDirectoryPath - The path to the repository.
+ * @returns The default branch name.
+ * @throws If neither "main" nor "master" exist.
+ */
+export async function getDefaultBranchName(
+  repositoryDirectoryPath: string,
+): Promise<string> {
+  try {
+    log('Running: git rev-parse --verify --quiet main');
+    await execa('git', ['rev-parse', '--verify', '--quiet', 'main'], {
+      cwd: repositoryDirectoryPath,
+    });
+    return 'main';
+  } catch (error) {
+    log('Command `git rev-parse --verify --quiet main` failed:', error);
+  }
+
+  try {
+    log('Running: git rev-parse --verify --quiet master');
+    await execa('git', ['rev-parse', '--verify', '--quiet', 'master'], {
+      cwd: repositoryDirectoryPath,
+    });
+    return 'master';
+  } catch (error) {
+    log('Command `git rev-parse --verify --quiet master` failed:', error);
+  }
+
+  throw new Error(
+    `Could not detect default branch name for repository '${repositoryDirectoryPath}'.`,
+  );
+}
+
+/**
+ * Retrieves the date/time that any commits were last fetched for the given
+ * repository by checking the modification time of `.git/FETCH_HEAD`.
+ *
+ * @param repositoryDirectoryPath - The path to the repository.
+ * @returns The date of the last fetch if it has occurred, or null otherwise.
+ */
+export async function getLastFetchedDate(
+  repositoryDirectoryPath: string,
+): Promise<Date | null> {
+  const stats = await getEntryStats(
+    path.join(repositoryDirectoryPath, '.git', 'FETCH_HEAD'),
+  );
+  return stats ? stats.mtime : null;
+}
+
+/**
+ * Ensures that the repository has fresh commits (where fresh means one hour or
+ * younger).
+ *
+ * @param repositoryDirectoryPath - The path to the repository.
+ * @param lastFetchedDate - The date/time when the repository
+ * was last fetched.
+ * @returns The last fetched date if it has been an hour or less, or now
+ * otherwise.
+ */
+export async function ensureDefaultBranchIsUpToDate(
+  repositoryDirectoryPath: string,
+  lastFetchedDate: Date | null,
+) {
+  const now = new Date();
+  if (
+    lastFetchedDate &&
+    now.getTime() - lastFetchedDate.getTime() <= ONE_HOUR
+  ) {
+    return lastFetchedDate;
+  }
+
+  log('Running: git pull');
+  await execa('git', ['pull'], {
```

Comment:
Nit: It might be simpler to use `fit fetch && git reset --hard origin/main` here, or something along those lines. `git pull` will merge the latest main branch into the local branch, which is not really what we want to do here. Git reset also means that we don't need the main branch to be checked out in the first place, so we no longer need the `getCurrentBranchName` step

Reply:
Good point. Updated here: [7d1ec62](https://github.com/MetaMask/module-lint/pull/30/commits/7d1ec62fbd9a88aa1d7fffe894a7c881eb802325)

Reply:
>  Git reset also means that we don't need the main branch to be checked out in the first place, so we no longer need the `getCurrentBranchName` step

This is updated here: [ced47f7](https://github.com/MetaMask/module-lint/pull/30/commits/ced47f70de8e140e00e6fe85db17af4d487e0bfc)